jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/component/Dumpable.java

```diff
@@ -345,4 +350,63 @@ default boolean isDumpable(Object o)
             return true;
         }
     }
+
+    /**
+     * An Appendable that helps avoid dump cycles.
+     */
+    class DumpAppendable implements Appendable
+    {
+        private final Appendable _appendable;
+        private final IdentityHashMap<Object, Boolean> _visited = new IdentityHashMap<>();
```

Comment:
You should use `Set<Object> _visited = Collections.newSetFromMap(new IdentityHashMap<>())` to present the map as a set.